packages/dci-api/src/async-search/async-search-handler.ts

```diff
@@ -15,22 +15,26 @@ import {
 import { parseToken } from '../auth'
 import { randomUUID } from 'node:crypto'
 import { type ReqResWithAuthorization } from '../server'
+import { generateSignature } from '../crypto/sign'
 
 async function asyncSearch(
   token: string,
   request: AsyncSearchRequest,
   correlationId: ReturnType<typeof randomUUID>
 ) {
   const results = await search(token, request.message)
+  const syncSearchResponse = registrySyncSearchBuilder(
+    results,
+    request,
+    correlationId
+  ) satisfies operations['post_reg_on-search']['requestBody']['content']['application/json']
+  syncSearchResponse.signature = await generateSignature({
+    header: syncSearchResponse.header,
+    message: syncSearchResponse.message
+  })
```

Comment:
Could we prevent mutating the response here some way.. 🤔

Reply:
hm we could just create another object

Reply:
what if the signature was made in dci-opencrvs-bridge in registrySyncSearchBuilder? 
Then architecturally we'd have like:
```
dci-api = input, output
dci-opencrvs-bridge = conversion, signature
opencrvs-api = calls to OpenCRVS
```
🤔